gson/src/test/java/com/google/gson/internal/bind/TypeAdaptersTestCase.java

```diff
@@ -0,0 +1,63 @@
+package com.google.gson.internal.bind;
+
+import com.diffblue.deeptestutils.Reflector;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.internal.bind.TypeAdapters;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class TypeAdaptersTestCase {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  /* testedClasses: com/google/gson/internal/bind/TypeAdapters.java */
+  /*
+   * Test generated by Diffblue Deeptest.
+   * This test case covers the entire method.
+   */
+
+  @Test
+  public void newFactoryForMultipleTypesInputNullNullNullOutputNotNull() {
+
+    // Arrange
+    final Class base = null;
+    final Class sub = null;
+    final TypeAdapter typeAdapter = null;
+
+    // Act
+    final TypeAdapterFactory retval =
+        TypeAdapters.newFactoryForMultipleTypes(base, sub, typeAdapter);
+
+    // Assert result
+    Assert.assertNotNull(retval);
+    Assert.assertEquals(null, Reflector.getInstanceField(retval, "val$typeAdapter"));
```

Comment:
This is not super-useful, unless there is also a test that checks that non-null values are correctly initialised too.

Reply:
I don't see an equivalent test for non-null values.